Flink job execution failed

Hi dataArtisans,

I'm trying to run the flink benchmark (AdvertisingTopologyNative). I manually started Zookeeper, Kafka, Redis and Flink job manager using:
```
./stream-bench.sh START_ZK
./stream-bench.sh START_KAFKA
./stream-bench.sh START_REDIS
./stream-bench.sh START_FLINK
```
Then I started the flink job using `./stream-bench.sh START_FLINK_PROCESSING` and after some time the job switches to Failed. Could you please help?

Here is the full output:
```
ubuntu@my-bench:~/yahoo-streaming-benchmark$ ./stream-bench.sh START_FLINK_PROCESSING
[2016-05-23 10:40:58,041] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
ubuntu@my-bench:~/yahoo-streaming-benchmark$ 05/23/2016 10:40:59        Job execution switched to status RUNNING.
05/23/2016 10:40:59     Source: Custom Source -> Flat Map(1/1) switched to SCHEDULED
05/23/2016 10:40:59     Source: Custom Source -> Flat Map(1/1) switched to DEPLOYING
05/23/2016 10:40:59     Flat Map -> Filter -> Projection -> Flat Map(1/1) switched to SCHEDULED
05/23/2016 10:40:59     Flat Map -> Filter -> Projection -> Flat Map(1/1) switched to DEPLOYING
05/23/2016 10:40:59     Flat Map(1/1) switched to SCHEDULED
05/23/2016 10:40:59     Flat Map(1/1) switched to DEPLOYING
05/23/2016 10:41:00     Source: Custom Source -> Flat Map(1/1) switched to RUNNING
05/23/2016 10:41:00     Flat Map(1/1) switched to RUNNING
05/23/2016 10:41:00     Flat Map -> Filter -> Projection -> Flat Map(1/1) switched to RUNNING
05/23/2016 10:42:01     Flat Map -> Filter -> Projection -> Flat Map(1/1) switched to CANCELED
05/23/2016 10:42:01     Source: Custom Source -> Flat Map(1/1) switched to FAILED
java.lang.NumberFormatException: For input string: "2181null"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Integer.parseInt(Integer.java:580)
        at java.lang.Integer.parseInt(Integer.java:615)
        at org.apache.zookeeper.client.ConnectStringParser.<init>(ConnectStringParser.java:72)
        at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:443)
        at org.apache.flink.shaded.org.apache.curator.utils.DefaultZookeeperFactory.newZooKeeper(DefaultZookeeperFactory.java:29)
        at org.apache.flink.shaded.org.apache.curator.framework.imps.CuratorFrameworkImpl$2.newZooKeeper(CuratorFrameworkImpl.java:146)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder$1.getZooKeeper(HandleHolder.java:94)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.internalClose(HandleHolder.java:128)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.closeAndReset(HandleHolder.java:77)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.reset(ConnectionState.java:217)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.checkTimeouts(ConnectionState.java:193)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.getZooKeeper(ConnectionState.java:87)
        at org.apache.flink.shaded.org.apache.curator.CuratorZookeeperClient.getZooKeeper(CuratorZookeeperClient.java:115)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper$1.call(EnsurePath.java:148)
        at org.apache.flink.shaded.org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper.ensure(EnsurePath.java:141)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath.ensure(EnsurePath.java:99)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsetFromZooKeeper(ZookeeperOffsetHandler.java:126)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsets(ZookeeperOffsetHandler.java:95)
        at org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer08.open(FlinkKafkaConsumer08.java:284)
        at org.apache.flink.api.common.functions.util.FunctionUtils.openFunction(FunctionUtils.java:38)
        at org.apache.flink.streaming.api.operators.AbstractUdfStreamOperator.open(AbstractUdfStreamOperator.java:91)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.openAllOperators(StreamTask.java:317)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:215)
        at org.apache.flink.runtime.taskmanager.Task.run(Task.java:559)
        at java.lang.Thread.run(Thread.java:745)

05/23/2016 10:42:01     Job execution switched to status FAILING.
java.lang.NumberFormatException: For input string: "2181null"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Integer.parseInt(Integer.java:580)
        at java.lang.Integer.parseInt(Integer.java:615)
        at org.apache.zookeeper.client.ConnectStringParser.<init>(ConnectStringParser.java:72)
        at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:443)
        at org.apache.flink.shaded.org.apache.curator.utils.DefaultZookeeperFactory.newZooKeeper(DefaultZookeeperFactory.java:29)
        at org.apache.flink.shaded.org.apache.curator.framework.imps.CuratorFrameworkImpl$2.newZooKeeper(CuratorFrameworkImpl.java:146)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder$1.getZooKeeper(HandleHolder.java:94)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.internalClose(HandleHolder.java:128)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.closeAndReset(HandleHolder.java:77)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.reset(ConnectionState.java:217)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.checkTimeouts(ConnectionState.java:193)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.getZooKeeper(ConnectionState.java:87)
        at org.apache.flink.shaded.org.apache.curator.CuratorZookeeperClient.getZooKeeper(CuratorZookeeperClient.java:115)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper$1.call(EnsurePath.java:148)
        at org.apache.flink.shaded.org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper.ensure(EnsurePath.java:141)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath.ensure(EnsurePath.java:99)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsetFromZooKeeper(ZookeeperOffsetHandler.java:126)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsets(ZookeeperOffsetHandler.java:95)
        at org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer08.open(FlinkKafkaConsumer08.java:284)
        at org.apache.flink.api.common.functions.util.FunctionUtils.openFunction(FunctionUtils.java:38)
        at org.apache.flink.streaming.api.operators.AbstractUdfStreamOperator.open(AbstractUdfStreamOperator.java:91)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.openAllOperators(StreamTask.java:317)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:215)
        at org.apache.flink.runtime.taskmanager.Task.run(Task.java:559)
        at java.lang.Thread.run(Thread.java:745)
05/23/2016 10:42:01     Flat Map(1/1) switched to CANCELING
05/23/2016 10:42:01     Flat Map(1/1) switched to CANCELED
05/23/2016 10:42:01     Job execution switched to status FAILED.

------------------------------------------------------------
 The program finished with the following exception:

org.apache.flink.client.program.ProgramInvocationException: The program execution failed: Job execution failed.
        at org.apache.flink.client.program.Client.runBlocking(Client.java:381)
        at org.apache.flink.client.program.Client.runBlocking(Client.java:355)
        at org.apache.flink.streaming.api.environment.StreamContextEnvironment.execute(StreamContextEnvironment.java:65)
        at org.apache.flink.streaming.api.environment.StreamExecutionEnvironment.execute(StreamExecutionEnvironment.java:1170)
        at flink.benchmark.AdvertisingTopologyNative.main(AdvertisingTopologyNative.java:69)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:505)
        at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:403)
        at org.apache.flink.client.program.Client.runBlocking(Client.java:248)
        at org.apache.flink.client.CliFrontend.executeProgramBlocking(CliFrontend.java:866)
        at org.apache.flink.client.CliFrontend.run(CliFrontend.java:333)
        at org.apache.flink.client.CliFrontend.parseParameters(CliFrontend.java:1189)
        at org.apache.flink.client.CliFrontend.main(CliFrontend.java:1239)
Caused by: org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
        at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply$mcV$sp(JobManager.scala:716)
        at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply(JobManager.scala:662)
        at org.apache.flink.runtime.jobmanager.JobManager$$anonfun$handleMessage$1$$anonfun$applyOrElse$7.apply(JobManager.scala:662)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:401)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.NumberFormatException: For input string: "2181null"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Integer.parseInt(Integer.java:580)
        at java.lang.Integer.parseInt(Integer.java:615)
        at org.apache.zookeeper.client.ConnectStringParser.<init>(ConnectStringParser.java:72)
        at org.apache.zookeeper.ZooKeeper.<init>(ZooKeeper.java:443)
        at org.apache.flink.shaded.org.apache.curator.utils.DefaultZookeeperFactory.newZooKeeper(DefaultZookeeperFactory.java:29)
        at org.apache.flink.shaded.org.apache.curator.framework.imps.CuratorFrameworkImpl$2.newZooKeeper(CuratorFrameworkImpl.java:146)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder$1.getZooKeeper(HandleHolder.java:94)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.internalClose(HandleHolder.java:128)
        at org.apache.flink.shaded.org.apache.curator.HandleHolder.closeAndReset(HandleHolder.java:77)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.reset(ConnectionState.java:217)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.checkTimeouts(ConnectionState.java:193)
        at org.apache.flink.shaded.org.apache.curator.ConnectionState.getZooKeeper(ConnectionState.java:87)
        at org.apache.flink.shaded.org.apache.curator.CuratorZookeeperClient.getZooKeeper(CuratorZookeeperClient.java:115)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper$1.call(EnsurePath.java:148)
        at org.apache.flink.shaded.org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath$InitialHelper.ensure(EnsurePath.java:141)
        at org.apache.flink.shaded.org.apache.curator.utils.EnsurePath.ensure(EnsurePath.java:99)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsetFromZooKeeper(ZookeeperOffsetHandler.java:126)
        at org.apache.flink.streaming.connectors.kafka.internals.ZookeeperOffsetHandler.getOffsets(ZookeeperOffsetHandler.java:95)
        at org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer08.open(FlinkKafkaConsumer08.java:284)
        at org.apache.flink.api.common.functions.util.FunctionUtils.openFunction(FunctionUtils.java:38)
        at org.apache.flink.streaming.api.operators.AbstractUdfStreamOperator.open(AbstractUdfStreamOperator.java:91)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.openAllOperators(StreamTask.java:317)
        at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:215)
        at org.apache.flink.runtime.taskmanager.Task.run(Task.java:559)
        at java.lang.Thread.run(Thread.java:745)
18401:M 23 May 11:43:06.038 * 100 changes in 300 seconds. Saving...
18401:M 23 May 11:43:06.040 * Background saving started by pid 18982
18982:C 23 May 11:43:06.089 * DB saved on disk
18982:C 23 May 11:43:06.090 * RDB: 6 MB of memory used by copy-on-write
18401:M 23 May 11:43:06.140 * Background saving terminated with success
```

Thank you.

--- a/stream-bench.sh
+++ b/stream-bench.sh
@@ -106,20 +106,43 @@ run() {
 
     echo 'kafka.brokers:' > $CONF_FILE
     echo '    - "localhost"' >> $CONF_FILE
+    echo 'kafka.port: 9092' >> $CONF_FILE
+    echo 'kafka.topic: "'$TOPIC'"' >> $CONF_FILE
+    echo 'kafka.partitions: '$PARTITIONS >> $CONF_FILE
+    echo 'kafka.zookeeper.path: /' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo 'akka.zookeeper.path: /akkaQuery' >> $CONF_FILE
     echo >> $CONF_FILE
     echo 'zookeeper.servers:' >> $CONF_FILE
     echo '    - "'$ZK_HOST'"' >> $CONF_FILE
+    echo 'zookeeper.port: '$ZK_PORT >> $CONF_FILE
     echo >> $CONF_FILE
-    echo 'kafka.port: 9092' >> $CONF_FILE
-	echo 'zookeeper.port: '$ZK_PORT >> $CONF_FILE
-	echo 'redis.host: "localhost"' >> $CONF_FILE
-	echo 'kafka.topic: "'$TOPIC'"' >> $CONF_FILE
-	echo 'kafka.partitions: '$PARTITIONS >> $CONF_FILE
-	echo 'process.hosts: 1' >> $CONF_FILE
-	echo 'process.cores: 4' >> $CONF_FILE
-	echo 'storm.workers: 1' >> $CONF_FILE
-	echo 'storm.ackers: 2' >> $CONF_FILE
-	echo 'spark.batchtime: 2000' >> $CONF_FILE
+    echo 'redis.host: "localhost"' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo 'process.hosts: 1' >> $CONF_FILE
+    echo 'process.cores: 4' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo '#STORM Specific' >> $CONF_FILE
+    echo 'storm.workers: 1' >> $CONF_FILE
+    echo 'storm.ackers: 2' >> $CONF_FILE
+    echo 'storm.highcard.redisthreads: 20' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo '#Spark Specific' >> $CONF_FILE
+    echo 'spark.batchtime: 2000' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo '#Flink Specific' >> $CONF_FILE
+    echo 'group.id: "flink_yahoo_benchmark"' >> $CONF_FILE
+    echo 'flink.checkpoint.interval: 60000' >> $CONF_FILE
+    echo 'add.result.sink: 1' >> $CONF_FILE
+    echo 'flink.highcard.checkpointURI: "file:///tmp/checkpoints"' >> $CONF_FILE
+    echo 'redis.threads: 20' >> $CONF_FILE
+    echo >> $CONF_FILE
+    echo '#EventGenerator' >> $CONF_FILE
+    echo 'use.local.event.generator: 1' >> $CONF_FILE
+    echo 'redis.flush: 1' >> $CONF_FILE
+    echo 'redis.db: 0' >> $CONF_FILE
+    echo 'load.target.hz: 10000000' >> $CONF_FILE
+    echo 'num.campaigns: 1000000' >> $CONF_FILE
 	
     $MVN clean install -Dspark.version="$SPARK_VERSION" -Dkafka.version="$KAFKA_VERSION" -Dflink.version="$FLINK_VERSION" -Dstorm.version="$STORM_VERSION" -Dscala.binary.version="$SCALA_BIN_VERSION" -Dscala.version="$SCALA_BIN_VERSION.$SCALA_SUB_VERSION"
 

